frontend/src/core/interceptors/global-error-handler.ts

```diff
@@ -1,34 +1,52 @@
 import { HttpErrorResponse } from '@angular/common/http';
-import { ErrorHandler, Inject, Injector, Injectable } from '@angular/core';
+import { ErrorHandler, Injector, Injectable } from '@angular/core';
 import { ToastrService } from 'ngx-toastr';
+import { ToastrConfig } from 'src/entity-models/const-resources/toastr-config';
+import { AuthService } from '../services/auth.service';
 
 @Injectable()
 export class GlobalErrorHandler extends ErrorHandler {
-  constructor(@Inject(Injector) private injector: Injector) {
+  constructor(
+    private injector: Injector) {
     super();
   }
 
   private get toastrService(): ToastrService {
     return this.injector.get(ToastrService);
   }
 
-  handleError(error: any): void {
-    if (
-      error instanceof HttpErrorResponse &&
-      error.status >= 400 &&
-      error.status < 500 &&
-      typeof error.error === 'string'
-    ) {
-      this.toastrService.error(error.error, '');
-    } else {
-      this.toastrService.error('An unexpected error has occurred.', 'Error', {
-        closeButton: true,
-        timeOut: 5000,
-        onActivateTick: true,
-      });
-    }
+  private get authService(): AuthService {
+    return this.injector.get(AuthService);
+  }
 
+  handleError(error: unknown): void {
     // TODO: Remove in prod
     super.handleError(error);
+
+    if (!(error instanceof HttpErrorResponse)) {
+      this.toastrService.error('An unexpected client-side error has occurred.', 'Error', ToastrConfig);
```

Comment:
config usage is defined in app.component ToatrService.forRoot, we don't need to even use it here since it works globally

Reply:
Done